### PR TITLE
Added context aware bibliography lookup

### PR DESCRIPTION
**Pull request by mehalter**
I love this plugin and wanted it to be able to contextually figure out where the bibliography file is and only open the bibliography that is relevant to the file that I am currently editing. This aims to do just that. Works great if I am writing a markdown or LaTeX file that references a bibliography in another folder.

Let me know if there is anything I need to change in functionality/documentation/formatting since this is my first contribution to the project.

### CODE REVIEW

**Comment by noahares** [comment]
Thanks for your Contribution! This sounds like a good feature to have and gets rid of the clunky multiple global bib-files. 

Because the features of this plugin start to accumulate I would prefer to move different functionality to different files. So could you move the helper functions to either `bibtex/utils.lua` or a new file? For the formatting just run it through `stylua` with the format file in the repo. Regarding the documentation I think something got mixed up there (see my comments).

I will try to break your code tomorrow and if I fail we can merge this ;)

File: README.md [hunk 1]
```diff
@@ -83,11 +83,45 @@ require"telescope".setup {
       -- Max number of authors to write in the formatted citation
       -- following authors will be replaced by "et al."
       citation_max_auth = 2,
+      -- Use context awareness by default
```
Comment (noahares) [inline README.md:86]: Context awareness is disabled by default

File: README.md [hunk 2]
```diff
@@ -83,11 +83,45 @@ require"telescope".setup {
...
+Two common contexts are included:
+
+| Filetype              | Context                                    |
+| --------------------- | ------------------------------------------ |
+| `pandoc`, `md`, `rmd` | `bibliography: file_path_with_ext`         |
+| `tex`                 | `\bibliography{relative_file_path_no_ext}` |
+
+_Note:_ Context awareness ignores the global bibliography files as well as the normal searching in the current directory.
+
+Context awareness can be enabled by default in the setup with `context`.
```
Comment (noahares) [inline README.md:104]: I think you should cut out the 'by default'.

**Comment by mehalter** [comment]
Awesome! I'll do these revisions tomorrow hopefully, might be Monday. One thing it doesn't cover atm is multibib. Also the code I have correctly verifies home dir with ~ but the function for getting the bib files does not. I haven't had a chance to try and debug that function, seems to be a problem with the `fs_stat` call

**Comment by mehalter** [comment]
I have ideas to add the multibib for both Pandoc and latex. The latex will be easier the Pandoc will be slightly more complicated because yaml lists have a few different possible formats

File: README.md [hunk 3]
```diff
@@ -83,11 +83,45 @@ require"telescope".setup {
...
+      -- Use context awareness by default
+      context = false
     },
   }
 }
 ```
 
+### Context Aware Bibliography File
+
+Two common contexts are included:
```
Comment (noahares) [inline README.md:95]: Maybe its just me, but it took a while to understand what context aware means. Maybe add something like: 'If enabled, the plugin will look for context lines in your currently opened file that imply which bibliography file you want to use. See bellow for common examples based on filetype:'

**Comment by noahares** [comment]
> Awesome! I'll do these revisions tomorrow hopefully, might be Monday. One thing it doesn't cover atm is multibib. Also the code I have correctly verifies home dir with ~ but the function for getting the bib files does not. I haven't had a chance to try and debug that function, seems to be a problem with the `fs_stat` call

Take your time. 
I do not understand the home directory part. Is this a problem with the current code? How does this show?

**Comment by mehalter** [comment]
Yeah this is a problem with the current code. It wasn't a case that would have ever come up though with the current code. The easiest way to test is to set up a Pandoc document and set the bibliography to a direct path that uses `~` in it. Then use this code and try to open the bibliography picker.

Something like

`bibliography: ~/refs.bib`

It will fail to load. It looks like the `fs_stat` command used in the function to load the bib files doesn't support expanding the `~` in a path.

**Comment by mehalter** [comment]
https://github.com/nvim-telescope/telescope-bibtex.nvim/blob/a2c14c84a493f20866cecf42db15ec3a3f84d42b/lua/telescope/_extensions/bibtex.lua#L138

This line fails

File: README.md (hunk 1 above)
Comment (mehalter) [reply README.md]: Fixed in latest commit

File: README.md (hunk 2 above)
Comment (mehalter) [reply README.md]: Fixed in latest commit

File: README.md (hunk 3 above)
Comment (mehalter) [reply README.md]: Fixed in latest commit

**Comment by mehalter** [comment]
Also let me know if you would prefer me move these to a new util file maybe `context.lua` or something that just has the parsing functions for different file types

**Comment by mehalter** [comment]
Ok @noahares  I just finished adding support for multiple bibliography files for both Pandoc and LaTeX. There are a few different supported syntaxes:

Pandoc:

```yaml
bibliography: refs1.bib
```

```yaml
bibliography: [refs1.bib, refs2.bib]
```

```yaml
bibliography:
  - refs1.bib
  - refs2.bib
```

LaTeX:

```tex
\bibliography{refs1}
```

```tex
\bibliography{refs1,ref2}
```

```tex
\addbibresource{refs1.bib}
\addbibresource{refs2.bib}
```

**Comment by mehalter** [comment]
Ah I didn't think about that, yeah I agree. What if we put a config variable as well for enabling or disabling the falling back logic? @noahares

**Comment by noahares** [comment]
I cannot think of a case where I would want to see nothing instead of the fallback (otherwise why would I have invoked telescope in the first place?). So I don't think we need a toggle for that.

**Comment by mehalter** [comment]
Yeah I guess I was thinking someone may want invoking telescope and seeing nothing to be an indication that I forgot to put a bibliography entry if one tends to not work in multi-file latex documents.

**Comment by noahares** [comment]
Great! The setting is fine if the fallback is enabled by default. Small typo, then I will merge this.

**Pull request merged**